src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java

```diff
@@ -122,9 +126,34 @@ private static void handlePlanTripResponse(Request request, OtpDispatcherRespons
         long tripStorageStartTime = DateTimeUtils.currentTimeMillis();
 
         Auth0Connection.checkUser(request);
-        Auth0UserProfile profile = Auth0Connection.getUserFromRequest(request);
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(request);
+        // A requesting user (Otp or third party user) is required to proceed.
+        if (requestingUser.otpUser == null && requestingUser.apiUser == null) {
+            return;
+        }
+        // TODO: Consider moving/replicating these checks (or a subset of) to before sending the request to OTP.
+        // If a user id is provided, the assumption is that an Api user is making a plan request on behalf of an Otp user.
+        String userId = request.queryParams(USER_ID_PARAM);
+        OtpUser otpUser = null;
+        if (requestingUser.otpUser != null && userId == null) {
+            // Otp user making a trip request for self.
+            otpUser = requestingUser.otpUser;
+        } else if (requestingUser.apiUser != null) {
+            // Api user making a trip request on behalf of an Otp user. In this case, the Otp user id must be provided
+            // as a query parameter.
+            otpUser = Persistence.otpUsers.getById(userId);
+            if (otpUser == null) {
```

Comment:
My bad, add `&& userId != null`, so that the `ApiUser` can still make OTP requests without specifying a user id.

Reply:
Updated.